### DISCUSSION TITLE
The 0.2.1-RC version makes 6-ways intersection's traffic lights worse

### DISCUSSION THREAD

**Discussion by GIS-PuppetMaster** [slyh/Cities2-TrafficLightsEnhancement#49] [General]
In the previous version, the 3-ways intersection rule seems to be working for 6-ways intersections, but the 0.2.1-RC can only manage intersections with equal or less than 4-ways

**Comment by ColdCathode01** [comment]
I do believe the mod is only "technically" working with 3 and 4 way intersections at the moment. As of 0.2.1-RC the mod detects the type of intersection and gives options based on that. It worked in previous versions, but I don't think that was intended, the sort of "bug as a feature" phenomena, since it just showed all options for any intersection.

**Comment by GIS-PuppetMaster** [comment]
> I do believe the mod is only "technically" working with 3 and 4 way intersections at the moment. As of 0.2.1-RC the mod detects the type of intersection and gives options based on that. It worked in previous versions, but I don't think that was intended, the sort of "bug as a feature" phenomena, since it just showed all options for any intersection.

Of course you are right. However, the old 6-ways intersections are more efficient than the vanilla rules.

**Comment by slyh** [comment]
I'm not sure what kind of six-way junction you are referring to, but starting from version 0.2.1-RC, it does limit the number of patterns that can be chosen by the player. I have introduced this change with the goal of reducing confusion, and it could be seen as a feature downgrade in some ways.

However, I also think it's impossible to provide presets for every junction layout that exists. In the long term, some sort of custom signal sequence will be needed.

**Reply by GIS-PuppetMaster** [reply]
The "6-way" junction means that the junction when you have 3 roads crossing in one single node. The newest version works fine for me.
